### ISSUE TITLE
WebXRCameraSet (WebXR.WebXRManager) called every frame

### ISSUE THREAD

**Issue by TheXRMonk** [De-Panther/unity-webxr-export#141]
Just noticed this running in the browser console every frame - wondering if I'm doing something wrong?

Also getting this warning;
warning: 2 FS.syncfs operations in flight at once, probably just doing extra work

**Comment by De-Panther** [comment]
What happens every frame?

**Comment by TheXRMonk** [comment]
> What happens every frame?

![image](https://user-images.githubusercontent.com/54309738/115407894-b670c980-a1f0-11eb-9885-b23ecd5981d8.png)

Can't seem to find a console log anywhere.

**Comment by De-Panther** [comment]
can you show what is on line 4462 in the build.framework?

**Comment by TheXRMonk** [comment]
> can you show what is on line 4462 in the build.framework?

I'm sorry it's all compiled into .unityweb so can't read anything - still learning about WebGL development.

**Comment by De-Panther** [comment]
in the image you posted, the text on the right is a link to that line...

**Comment by TheXRMonk** [comment]
> in the image you posted, the text on the right is a link to that line...

Shit, my bad;
```

function _JS_Log_Dump(ptr, type) {
 var str = Pointer_stringify(ptr);
 if (typeof dump == "function") dump(str);
 switch (type) {
 case 0:
 case 1:
 case 4:
  console.error(str);
  return;
 case 2:
  console.warn(str);
  return;
 case 3:
 case 5:
  console.log(str);
  return;
 default:
  console.error("Unknown console message type!");
  console.error(str);
 }
}
```

line 4462 is `console.log(str);`

Any way to get the stacktrace?

**Comment by De-Panther** [comment]
are there any logs before/after this line from the screen capture? is there an arrow on the right of this message?

**Comment by TheXRMonk** [comment]
> are there any logs before/after this line from the screen capture? is there an arrow on the right of this message?

A few (non-critical) warnings about standard unity editor serialization stuff 
not after.

Here's a link to the latest build in case you want to see it for yourself (not a development build, so no debug symbols);
https://virsabi.github.io/ScottWebXRDemo/

Thanks for ELI5 me through this!

**Comment by De-Panther** [comment]
Happy to help, but please, can you share those warnings, instead of writing that there are warnings? it can save us time and posts. when asking for log, it helps when you share what possible and not just a part of a line.
you can hide personal data from the log if there is such data

**Comment by TheXRMonk** [comment]
> Happy to help, but please, can you share those warnings, instead of writing that there are warnings, it can save us time and posts. when asking for log, it helps when you share what possible and not just a part of a line.
> you can hide personal data from the log if there is such data

These warnings,
`The referenced script on this Behaviour (Game Object 'UIFixPoints') is missing!`
are what I consider non-critical unity warnings - these come from errors like this;
```
A scripted object (probably AV.Hierarchy.Collection?) has a different serialization layout when loading. (Read 32 bytes but expected 40 bytes)
Did you #ifdef UNITY_EDITOR a section of your serialized properties in any of your scripts?
```

The only other warnings I have are one of these;
`[Deprecation] SharedArrayBuffer will require cross-origin isolation as of M91, around May 2021. See https://developer.chrome.com/blog/enabling-shared-array-buffer/ for more details.`
and three identical of these:
`WebGL: INVALID_ENUM: getInternalformatParameter: invalid internalformat`

Here is the whole log;
https://pastebin.com/s9nRAgA5

**Comment by De-Panther** [comment]
referenced script might be related to this issue.

Does the demo also have this repeating message?
https://de-panther.github.io/unity-webxr-export

I'll try to run your build later. Not near a computer now

**Comment by TheXRMonk** [comment]
> referenced script might be related to this issue.
> 
> Does the demo also have this repeating message?
> https://de-panther.github.io/unity-webxr-export
> 
> I'll try to run your build later. Not near a computer now

Yea running the demo I do not get the repeating message.

**Comment by TheXRMonk** [comment]
![image](https://user-images.githubusercontent.com/54309738/115551506-77ec1500-a2ab-11eb-9420-a0cefeb6a293.png)
Just a quick update, I tried a build without the smart-hierarchy collections, and I seem to still have the log calling camera set every frame. 
I'll do some more digging and update you if I find something.

**Comment by De-Panther** [comment]
What Unity version and what WebXR Export version?

**Comment by TheXRMonk** [comment]
> What Unity version and what WebXR Export version?

Unity 2020.2.5, WebXR Export 0.8.1

I'll try updating the exporter to see if that solves it

EDIT: Did not solve it.

**Comment by De-Panther** [comment]
Sorry, missed your comment last time.
It doesn't look like a WebXR Export issue, but something with your project.

If it still happens, you can try and build the project after setting the `emscripterArgs` to `--profiling-funcs`.
Add this in an editor script:
```
PlayerSettings.WebGL.emscriptenArgs = "--profiling-funcs";
```
